go/ql/lib/semmle/go/frameworks/Logrus.qll

```diff
@@ -49,4 +54,67 @@ module Logrus {
 
     override int getFirstFormattedParameterIndex() { result = argOffset + 1 }
   }
+
+  private class SetFormatterFunction extends Function {
+    SetFormatterFunction() {
+      this.hasQualifiedName(packagePath(), "SetFormatter") or
+      this.(Method).hasQualifiedName(packagePath(), "Logger", "SetFormatter")
+    }
+  }
+
+  private class LoggerFormatter extends Field {
+    LoggerFormatter() { this.hasQualifiedName(packagePath(), "Logger", "Formatter") }
+  }
+
+  private class JsonFormatter extends Type {
+    JsonFormatter() { this.hasQualifiedName(packagePath(), "JSONFormatter") }
+  }
+
+  /**
+   * Gets the types corresponding to sanitizing formatters.
+   */
+  private Type sanitizingFormatter() { result instanceof JsonFormatter }
```

Comment:
Am I right in thinking that users can define their own formatters? Do you think it's worth restructuring this to allow a user to add in this information, e.g. by having an abstract class `SanitizingFormatter` and making `JsonFormatter` extend/be an instance of it?